lax/lichens/sciencerun0.py

```diff
@@ -616,40 +617,49 @@ def _process(self, df):
             return df
 
 
-class S1AreaFractionTop(RangeLichen):
+class S1AreaFractionTop(StringLichen):
     '''S1 area fraction top cut
 
-    Uses scipy.stats.binom_test to compute a p-value based on the
+    Uses a modified version of scipy.stats.binom_test to compute a p-value based on the
     observed number of s1 photons in the top array, given the expected
-    probability that a photon at the event's (r,z) makes it to the top array.
+    probability that a photon at the event's (x,y,z) makes it to the top array.
+    Modifications made to original algorithm implemented in pax increase sensitivity for small s1s.
+    For pax < v6.6.0 computes p-value live, otherwise loads it from disk. Computation done here is
+    not as accurate as in pax > v6.6.5
+
+    Requires Extended minitrees >= v0.0.4
 
     Uses a 3D map generated with Kr83m 32 keV line
 
     note: https://xecluster.lngs.infn.it/dokuwiki/doku.php?id=xenon:xenon1t:darryl:xe1t_s1_aft_map
 
-    Author: Darryl Masson, dmasson@purdue.edu
+    Contact: Darryl Masson, dmasson@purdue.edu
     '''
 
-    version = 1
-    variable = 'pvalue_s1_area_fraction_top'
+    variable = 's1_area_fraction_top_probability'
     allowed_range = (1e-4, 1 + 1e-7)  # must accept p-value = 1.0 with a < comparison
 
     def __init__(self):
-        aftmap_filename = os.path.join(DATA_DIR,
-                                       's1_aft_rz_02Mar2017.json')
-        with open(aftmap_filename) as data_file:
-            data = json.load(data_file)
-        r_pts = np.array(data['r_pts'])
-        z_pts = np.array(data['z_pts'])
-        aft_vals = np.array([data['map'][i*len(z_pts):(i+1)*len(z_pts)] for i in range(len(r_pts))]) # unpack 1d array to 2d
-        self.aft_map = RectBivariateSpline(r_pts,z_pts,aft_vals)
+        self.min_pax_version = 660  # before v6.6.0 p-value not calculated
+        if int(pax_version.replace('.', '')) < self.min_pax_version:
```

Comment:
I found a better way to do version checking that didn't require asking about pax versions. This cut is going to get rewritten for SR1.